Engine/src/neat/neat-genome.cpp

```diff
@@ -584,6 +594,75 @@ void Genome::MutateActivationFunction() {
     neurons_[indexRandomNeuronHidden].SetActivation(activationTypes[randomIndex]);
 }
 
+/*!
+ * @brief Activates a random brain module from the available modules in the Genome.
+ *
+ * Adds a randomly selected brain module to the genome, updating input and output
+ * neurons of the module to reflect the current state of the genome.
+ */
+void Genome::MutateActivateBrainModule(){
+
+  std::random_device rd;
+  std::mt19937 generator(rd());
+  std::uniform_int_distribution<int> distribution(0, AvailableModules.size() - 1);
+  int randomIndex = distribution(generator);
+  BrainModule module = AvailableModules.at(randomIndex);
+  modules_.push_back(module);
+
+  int input_size = module.GetInputNeuronIds().size();
+  std::vector<int> input_ids(input_size, 0);
+  for (int i = 0; i < module.GetInputNeuronIds().size(); i++){
+    if (i == 0) module.SetFirstInputId(GetInputCount());
+    AddNeuron(Neuron(NeuronType::kInput, 0));
```

Comment:
adding neurons every time you activate a module can be very problematic. When a neural network is created from a genome, the order of input and output neurons in the neurons_ vector matters. Since modules are activated randomly through mutatons, we have no way of knowing which module is activated first, which is second etc. Thus modules' input neurons can be in different locations in different creatures' brains but input data is passed always in the same order. You see why this is a problem...